### ISSUE TITLE
Devices don't get enabled when writing configuration to device

### ISSUE THREAD

**Issue by fdanapfel** [rabbitholecomputing/SCSI2SD-util-V6#9]
Hello,

I'm trying to use version 1.0.4 of SCSI2SD-util-V6 from this repo on macOS Catalina 10.15.7 to configure a SCSI2SD v6 / 2020d card with firmware 6.3.1, since the original version provided by the creator of the SCSI2SD card doesn't work correctly on Catalina (the menu doesn't work).

I'm able to change all settings the way I want to, including configuring the devices, and write the configuration to the device. But when I attach the SCSI2SD v6 card to the system I want to use it on no devices appear.

When checking the configuration using the Linux version of SCSI2SD-util-v6 it shows the the devices which where shown as 'Enabled' in the macOS version of SCSI2SD-util-v6 as all being 'Disabled'.

This is also confirmed by saving the configuration to a file on macOS and then checking iit with a text editor. For example here is the configuration section for a 1GB hardisk with SCSI ID 0 which should have been enabled on the card:
```
<SCSITarget id="0">
    <enabled>false</enabled>

    <!-- ********************************************************
    Space separated list. Available options:
    apple		Returns Apple-specific mode pages
    omti		OMTI host non-standard link control
    xebec		XEBEC ignore step options in control byte
    ********************************************************* -->
    <quirks></quirks>


    <!-- ********************************************************
    0x0    Fixed hard drive.
    0x1    Removable drive.
    0x2    Optical drive  (ie. CD drive).
    0x3    1.44MB Floppy Drive.
    ********************************************************* -->
    <deviceType>0</deviceType>


    <!-- ********************************************************
    Device type modifier is usually 0x00. Only change this if your
    OS requires some special value.

    0x4C    Data General Micropolis disk
    ********************************************************* -->
    <deviceTypeModifier>0</deviceTypeModifier>


    <!-- ********************************************************
    SD card offset, as a sector number (always 512 bytes).
    ********************************************************* -->
    <sdSectorStart>0</sdSectorStart>


    <!-- ********************************************************
    Drive geometry settings.
    ********************************************************* -->

    <scsiSectors>2097151</scsiSectors>
    <bytesPerSector>512</bytesPerSector>
    <sectorsPerTrack>63</sectorsPerTrack>
    <headsPerCylinder>255</headsPerCylinder>


    <!-- ********************************************************
    Drive identification information. The SCSI2SD doesn't
    care what these are set to. Use these strings to trick a OS
    thinking a specific hard drive model is attached.
    ********************************************************* -->

    <!-- 8 character vendor string -->
    <!-- For Apple HD SC Setup/Drive Setup, use ' SEAGATE' -->
    <vendor> codesrc</vendor>

    <!-- 16 character produce identifier -->
    <!-- For Apple HD SC Setup/Drive Setup, use '          ST225N' -->
    <prodId>         SCSI2SD</prodId>

    <!-- 4 character product revision number -->
    <!-- For Apple HD SC Setup/Drive Setup, use '1.0 ' -->
    <revision> 6.0</revision>

    <!-- 16 character serial number -->
    <serial>12345678        </serial>

</SCSITarget>
```
Is there anything I can do to fix this?

**Comment by aperezbios** [comment]
@fdanapfel Thanks for the report. That's strange, and will be investigated ASAP. In the meantime, it _is_ possible to use the original scsi2sd-util6 utility on Catalina (x86_64/Intel only). The workaround is to de-focus the application once you've started it (you can hide it with Command-H) and then bring it back in to the foreground, at which time it should behave more or less correctly.

**Comment by gcasa** [comment]
@aperezbios @fdanapfel The issue here was a combination of things.   The scsiId is logically or'd with the flag that determines if the device is enabled.   For instance, if you have device = 0 and it is enabled the result would be 0x80 (the enabled flag is dec 128 (0x80) so 0x0 | 0x80 = 0x80).  Similarly, if you have device 1 (0x01) and it's enabled the flag will be 0x81.   The math to do this on the way out was not being done properly, which was screwing everything up.   I have corrected this.   Please try again and let me know if there are any issues.  Thx.  GC

**Comment by aperezbios** [comment]
> Is there anything I can do to fix this?


Nope, but please try this build. It isn't signed/notarized, so macOS will complain about that and you'll need to go in to your security Preference Pane to override it and let it run, but it should have the problem fixed. If you could confirm, that would be fantastic.
[SCSI2SD-util-v6.zip](https://github.com/rabbitholecomputing/SCSI2SD-util-V6/files/6379873/SCSI2SD-util-v6.zip)

**Comment by aperezbios** [comment]
Fixed in 15543fa

**Comment by aperezbios** [comment]
@fdanapfel, this has been fixed in [release version 1.0.5](https://github.com/rabbitholecomputing/SCSI2SD-util-V6/releases/tag/1.0.5). Thank you for the report.

**Comment by fdanapfel** [comment]
@aperezbios @gcasa Thanks for the replies and for providing a fix.

I just tested version [1.0.5](https://github.com/rabbitholecomputing/SCSI2SD-util-V6/releases/tag/1.0.5) (the 'About' box still says version 1.0.4 btw.) and now the devices get enabled correctly.
